css/css-transitions/non-rendered-element-004.tentative.html is a unique failure

graouts opened the pull request:
#### e48050be43a572745d972e1aff015538f469b9d4
<pre>
css/css-transitions/non-rendered-element-004.tentative.html is a unique failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=235137">https://bugs.webkit.org/show_bug.cgi?id=235137</a>
rdar://87785165

Reviewed by Antti Koivisto.

While we correctly call Styleable::cancelDeclarativeAnimations() during renderer teardown
in RenderTreeUpdater::tearDownRenderers(), only the provided element and its descendants
would be considered, but we did no work for anonymous renderers such as ::marker renderers.

We now also call Styleable::cancelDeclarativeAnimations() if we&apos;re dealing with an element
with a RenderListMarker renderer.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-transitions/non-rendered-element-004.tentative-expected.txt:
* Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
(WebCore::RenderTreeUpdater::tearDownRenderers):

Canonical link: <a href="https://commits.webkit.org/260352@main">https://commits.webkit.org/260352@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/237eea45645f7fbea19aeaa0633970a624cd47ef

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107936 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16993 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40820 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117069 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116416 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111823 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18380 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8311 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100124 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113699 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13874 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97092 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41693 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95786 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28729 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83440 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9917 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30077 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10624 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6973 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16068 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49666 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12196 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3894 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

anttijk commented on Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
```diff
@@ -610,6 +612,11 @@ void RenderTreeUpdater::tearDownRenderers(Element& root, TeardownType teardownTy
+                if (renderListItem->markerRenderer())
+                    Styleable(element, PseudoId::Marker).cancelDeclarativeAnimations();
+            }
+
```
Maybe these could go to a function, `tearDownPseudoElementRenderers()` or similar?

graouts committed:
css/css-transitions/non-rendered-element-004.tentative.html is a unique failure
https://bugs.webkit.org/show_bug.cgi?id=235137
rdar://87785165

Reviewed by Antti Koivisto.

While we correctly call Styleable::cancelDeclarativeAnimations() during renderer teardown
in RenderTreeUpdater::tearDownRenderers(), only the provided element and its descendants
would be considered, but we did no work for anonymous renderers such as ::marker renderers.

We now also call Styleable::cancelDeclarativeAnimations() if we're dealing with an element
with a RenderListMarker renderer.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-transitions/non-rendered-element-004.tentative-expected.txt:
* Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
(WebCore::RenderTreeUpdater::tearDownRenderers):

Canonical link: https://commits.webkit.org/260352@main
